src/main/java/org/spongepowered/common/event/tracking/phase/packet/AttackEntityPacketState.java

```diff
@@ -65,4 +88,98 @@ public void populateContext(EntityPlayerMP playerMP, Packet<?> packet, PhaseCont
+                final UUID key = entry.getKey();
+                final Optional<Entity> affectedEntity = spongeWorld.getEntity(key);
+                if (!affectedEntity.isPresent()) {
+                    continue;
+                }
+                final Collection<ItemDropData> itemStacks = entry.getValue();
+                if (itemStacks.isEmpty()) {
+                    return;
+                }
+                final List<ItemDropData> items = new ArrayList<>();
+                items.addAll(itemStacks);
+
+                if (!items.isEmpty()) {
+                    final List<Entity> itemEntities = items.stream()
+                            .map(data -> data.create(((WorldServer) player.world)))
+                            .map(EntityUtil::fromNative)
+                            .collect(Collectors.toList());
+                    final Cause cause = Cause.source(EntitySpawnCause.builder()
+                            .entity(affectedEntity.get())
+                            .type(InternalSpawnTypes.DROPPED_ITEM)
+                            .build()
+                    )
+                            .named(NamedCause.of("Attacker", player))
+                            .build();
+                    final DropItemEvent.Destruct
+                            destruct =
+                            SpongeEventFactory.createDropItemEventDestruct(cause, itemEntities);
+                    SpongeImpl.postEvent(destruct);
+                    if (!destruct.isCancelled()) {
+                        PacketPhaseUtil.processSpawnedEntities(player, destruct);
+                    }
+                }
+            }
+        });
+        context.getCapturedEntityItemDropSupplier().ifPresentAndNotEmpty(map -> {
+            for (Map.Entry<UUID, Collection<EntityItem>> entry : map.asMap().entrySet()) {
+                final UUID key = entry.getKey();
+                final Optional<Entity> attackedEntities = spongeWorld.getEntity(key);
+                if (!attackedEntities.isPresent()) {
+                    continue;
+                }
+                final List<Entity> items = entry.getValue().stream().map(EntityUtil::fromNative).collect(Collectors.toList());
+                final Cause cause = Cause.source(EntitySpawnCause.builder()
+                        .entity(EntityUtil.fromNative(player))
+                        .type(InternalSpawnTypes.DROPPED_ITEM)
+                        .build()
+                )
+                        .named(NamedCause.of("Attacker", player))
+                        .build();
+                final DropItemEvent.Destruct
+                        destruct =
+                        SpongeEventFactory.createDropItemEventDestruct(cause, items);
+                SpongeImpl.postEvent(destruct);
+                if (!destruct.isCancelled()) {
+                    PacketPhaseUtil.processSpawnedEntities(player, destruct);
```

Comment:
Another issue I have with the tracker is the sheer amount of scattered code. ALL THE UTILS need to be under one roof. Even if it means a massive ass class.

@JBYoshi @gabizou